### PR TITLE
Display: Fix for updating detail of an alert when submitting the end

### PR DESCRIPTION
**Pull request by mgbaybay**
## Changes
- Display: Update detail of an alert when submitting the end

Relates to: https://github.com/xibosignage/xibo/issues/3478

### CODE REVIEW

File: lib/Xmds/Soap.php [hunk 1]
```diff
@@ -3202,7 +3202,7 @@ private function createDisplayAlert(\DomElement $alertNode)
             $eventTypeId = $displayEvent->getEventIdFromString($eventType);
             empty($refId)
                 ? $displayEvent->eventEnd($this->display->displayId, $eventTypeId, $date)
-                : $displayEvent->eventEndByReference($this->display->displayId, $eventTypeId, $refId);
+                : $displayEvent->eventEndByReference($this->display->displayId, $eventTypeId, $refId, $detail);
```
Comment (dasgarner) [inline lib/Xmds/Soap.php:3205]: Is it possible to also do it if there is no reference?

File: lib/Entity/DisplayEvent.php [hunk 2]
```diff
@@ -165,8 +165,11 @@ public function eventEndByReference(int $displayId, int $eventTypeId, int $refId
             )
         );
 
+        // When updating the event end, concatenate the end message to the current message
         $this->getStore()->update(
-            'UPDATE `displayevent` SET `end` = :toDt
+            'UPDATE `displayevent` SET 
+                      `end` = :toDt, 
+                      `detail` = CONCAT_WS(". ", NULLIF(CONCAT_WS(".", NULLIF(`detail`, "")), ""), :detail)
```
Comment (dasgarner) [inline lib/Entity/DisplayEvent.php:172]: Can we use single quotes for consistency?

Comment (dasgarner) [reply lib/Entity/DisplayEvent.php:172]: Also can you explain why the double concat_ws in a comment so its easy to refer to later?

**Review by dasgarner — CHANGES_REQUESTED** [review]

**Review by dasgarner — APPROVED** [review]

**Comment by ifarzana** [comment]
TEST: /Display END

**Pull request merged**